infrasim/vm.py

```diff
@@ -101,115 +107,109 @@ def set_network(self, mode, name):
         nets = []
         change_flag = False
 
-        conf = cfgparse.ConfigParser()
-        f = conf.add_file(VM_DEFAULT_CONFIG, type = 'ini')
-        conf.parse()
+        with open(VM_DEFAULT_CONFIG, 'r') as f_yml:
+            conf = yaml.load(f_yml)
+
+        # conf = cfgparse.ConfigParser()
+        # f = conf.add_file(VM_DEFAULT_CONFIG, type = 'ini')
+        # conf.parse()
 
         # add existing options into config parser
-        if self.has_option(conf, "node", "qemu_mac_num") is True:
-            conf.add_option("qemu_mac_num", keys="node")
+
+        # if has_option(conf, "node", "qemu_mac_num") is True:
+        #     conf.add_option("qemu_mac_num", keys="node")
 
         mac_list = []
         for i in range(0, MAX_QEMU_MAC_NUM):
-            if self.has_option(conf, "node", "qemu_mac"+str(i)) is True:
-                mac_list.append(conf.add_option("qemu_mac"+str(i), keys="node"))
+            if has_option(conf, "node", "qemu_mac"+str(i)):
```

Comment:
typo? in yml, it is written in:
    network_mac1: {{mac1}}
    #network_mac2={{mac2}}
    #network_mac3={{mac3}}